### ISSUE TITLE
No longer attaches to existing session

### ISSUE THREAD

**Issue by Richard-Walton** [tmuxinator/tmuxinator#402]
This commit (https://github.com/tmuxinator/tmuxinator/commit/452d3bd6135631b58af9f75599b8899ebaec63d2) seems to generate an invalid command on my machine.  The commit is from Nov 2015(!) though, which seems super strange to start causing issues.  Has this code just made it into a build?

Anyway, the command (After removing the >/dev/null) fails with:

`usage: has-session [-t target-session]`

Causing the exit code to be 1.  This then means tmuxinator will create the windows/panes again every time :(

**Comment by Richard-Walton** [comment]
Ok - Further investigation reveals that the `has-session = ...` syntax is not available in tmux1.8 (The version I have installed from ubuntu/trusty repos)

I tried upgrading to tmux v2.0 - the problem still persists.

**Comment by Richard-Walton** [comment]
Downgrading to 0.7.1 fixes the issue

**Comment by J3RN** [comment]
@Richard-Walton Hey there! I'm sorry about this issue you've been having. The change was part of a feature/fix where tmuxinator would formerly attach by prefix -- so if you has a session named "foobar" and ran `mux foo`, you'd be attached to the "foobar" session.

I'm not sure why upgrading to 2.0 didn't help, but I'm currently running 2.1, so it may have been added in that release. In any event, I can imagine that Ubuntu users are probably a large portion of our user base, so getting this fixed is a priority.

In the tmux man page, I found:
> If the session name is prefixed with an '=', only an exact match is accepted (so '=mysess' will only match exactly 'mysess', not 'mysession').

Could you try running something like:
```bash
tmux has-session -t =foo
```
and post the output?

**Comment by Richard-Walton** [comment]
@J3RN - No problem.  Thanks for maintaining such a great tool.

With tmux 1.8 I have a session named "dev-env" running:

```
tmux ls
dev-env: 18 windows (created Mon Apr  4 12:51:57 2016) [166x51]

tmux has-session -t dev-env && echo $?
0
```

Unfortunately adding = as you suggested does not work:

```
tmux has-session -t =dev-env
session not found: =dev-env
```

**Comment by Richard-Walton** [comment]
I wonder if perhaps, to maintain compatibility with older versions of tmux, we could do something like:

`tmux -s | grep "^<% name %>:"`

?

**Comment by ethagnawl** [comment]
@Richard-Walton My apologies - I didn't realize that syntax wasn't backwards-compatible. 

Your proposed solution looks reasonable to me. Unfortunately, I'll be away from my computer for the rest of the week, so I'll be unable to help with testing/integration.

**Comment by Richard-Walton** [comment]
I've opened a pull request (https://github.com/tmuxinator/tmuxinator/pull/403)

I have a shameful admission to make:  I haven't been able to test this change outside of on my machine on the command line (using tmux 1.8).  

As long as I done the ruby template interpolation correctly (I'm not known for my ruby skills) and the `tmux ls` output hasn't changed since tmux 1.8 we're good to go.... :)

**Comment by caioariede** [comment]
Which version should I use to avoid this?

**Comment by caioariede** [comment]
Nvm, I got it. I installed the working version (0.7.1) but forgot to uninstall 0.7.2.

**Comment by ethagnawl** [comment]
Closed via #404.
